File: productExceptSelf.js
// https://neetcode.io/problems/products-of-array-discluding-self
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let ot = [];
        for(let i=0; i<nums.length; i++){
            let pro = 1;
            for(let j=0; j<nums.length; j++){
                if(i!=j){
                    pro *= nums[j];
                }
            }
            ot.push(pro);
        }
        return ot;
    }
}

File: productExceptSelf.py
// https://neetcode.io/problems/products-of-array-discluding-self
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        length = len(nums)
        result = [1] * length  # Initialize result array with 1

        # Calculate left products for each element
        left_product = 1
        for i in range(length):
            result[i] = left_product
            left_product *= nums[i]

        # Calculate right products and final output
        right_product = 1
        for i in range(length - 1, -1, -1):
            result[i] *= right_product
            right_product *= nums[i]

        return result